sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py

```diff
@@ -0,0 +1,451 @@
+                                  '2': {'min': 0,   'max': 90},
+                                  '1': {'min': 0,   'max': 90},
+                                  },
+                              'RF':
+                              {
+                                  '4': {'min': -20, 'max': 20},
+                                  '3': {'min': -15, 'max': 90},
+                                  '2': {'min': 0,   'max': 90},
+                                  '1': {'min': 0,   'max': 90},
+                                  },
+                              'LF':
+                              {
+                                  '5': {'min': 0,   'max': 45},
+                                  '4': {'min': -20, 'max': 20},
+                                  '3': {'min': -15, 'max': 90},
+                                  '2': {'min': 0,   'max': 90},
+                                  '1': {'min': 0,   'max': 90},
+                                  },
+                              'WR':
+                              {
+                                  '1': {'min': -40, 'max': 28},
+                                  '2': {'min': -30, 'max': 10},
+                                  },
+                              }
+
+        self._clear_j4 = {'FF': {'MF': 'max', 'RF': 'min', 'LF': 'min'},
+                          'MF': {'FF': 'min', 'RF': 'min', 'LF': 'min'},
+                          'RF': {'FF': 'min', 'MF': 'min', 'LF': 'min'},
+                          'LF': {'FF': 'min', 'MF': 'min', 'RF': 'max'}}
+        self._joint_state_subscriber = rospy.Subscriber('/joint_states', JointState, self.joint_state_cb)
+        self._hand_commander = SrHandCommander(name=(side + "_hand"))
+        for key, value in self._hand_commander.get_current_state().iteritems():
+            requested_joint = key.replace(self._hand_prefix, "")
+            self.requested_joints.append(requested_joint)
+        self._controller_helper = ControllerHelper([self._hand_prefix[0] + 'h'], [self._hand_prefix],
+                                                   [joint.lower() for joint in self.requested_joints])
+        self._hand_commander.move_to_joint_value_target(self._joint_states_zero, wait=True, angle_degrees=True)
+        self._switch_controller_service = rospy.ServiceProxy('controller_manager/switch_controller', SwitchController)
+        self._pwm_command_publishers = {}
+        self.setup_pwm_publishers()
+        self.initialise_output_dictionary()
+        self.setup_controller_subscribers()
+        while not rospy.is_shutdown():
+            self.run()
+
+    def run(self):
+        CONST_EXIT_CHAR = 'Q'
+        CONST_POSITION_CHAR = 'P'
+        CONST_EFFORT_CHAR = 'E'
+        self.print_help()
+        in_string = raw_input("Please enter command:").upper()
+        if in_string == "":
+            return
+        input_commnad = in_string.split('_')
+        command_type = input_commnad[0].upper()
+        if CONST_EXIT_CHAR in command_type:
```

Comment:
We should split this run function in multiple functions. it would be good to have e.g.  some of the if instead of like that passed through a funciton. and some of the for loops can also be wrap in functions